lib/sitemap-parser.rb

```diff
@@ -11,7 +11,11 @@ def initialize(url, opts = {})
   def raw_sitemap
     @raw_sitemap ||= begin
       if @url =~ /\Ahttp/i
-        request = Typhoeus::Request.new(@url, followlocation: @options[:followlocation])
+        request_options = { followlocation: @options[:followlocation] }
+        if @options[:userpwd]
+          request_options[:userpwd] = @options[:userpwd]
+        end
+        request = Typhoeus::Request.new(@url, request_options)
```

Comment:
Should we `delete` decurse before passing then and pass the rest? (Blacklist, rather than whitelist?)